### DISCUSSION TITLE
Should the current sample get any improvements?

### DISCUSSION THREAD

**Discussion by MasterAler** [MasterAler/SampleYUVRenderer#7] [Ideas]
I leave a discussion topic here just in case someone found the code useful and faces similar challenges in his coding activity.

Actually, once having to deal with video rendering in С++/Qt, a few recipes got collected and a few improvements were made some time ago when the current PoC became a production-ready component in a job project. 
What comes in mind instantly:
1. RGB rendering support, modern OpenGL usage with a fallback to older versions  
2. Wideowall mode, a set of subtextures forming a mosaic screen texture with multiple videos

Also, using FFMPEG decoding in a way it skips the `sws_scale` step was quite an edificatory piece of code.

Thus, if by chance someone would like to see some improvents, or maybe another repo with a more complex player demo , whatever -- feel free to share your ideas here, that's what this discussion is being left here for.

**Comment by rajatsrao** [comment]
Hi @MasterAler ,

I found this repo pretty useful. I had a similar GL renderer working in RGB domain. But I420 helps (lower CPU usage, lower memory usage). However, it's still not as good as I'd like because the layout I'm working with is NV12 which I need to first convert to I420.

Would you be able to add support for rendering NV12? Most HW accelerators work with NV12 and it would be good to directly render that.
Alternately, it would also help me to get rendering in YUY2 because most cameras output that.

**Reply by MasterAler** [reply]
Hi, @rajatsrao!

First of all, thanks for your feedback, glad to know someone's interested! 
Yeah, I faced a similar challenge once and would search for my old NV12 code, gonna do it at the weekend. I'll probably test it with some primitive ffmpeg sketch, will think of adding it here as well so that frame format output could be properly verified.

**Reply by rajatsrao** [reply]
I ended up doing this myself. It was pretty simple because your I420 renderer is written pretty well and already handles planar format.
I guess there were two main changes -

1. Remove the "V" part altogether. Change U from GL_LUMINANCE to GL_LUMINANCE_ALPHA (or on newer openGL version GL_RG). This will read 2 bytes into one pixel element.
2. Then in the conversion code, replace v.r with u.a (or on newer openGL version u.g)

**Reply by MasterAler** [reply]
Great! Glad that you've done it!
Anyways, feel free to contribute (if you want) and thanks again for your feedback, now I know, what would be useful to add to this sample.
